### PR TITLE
Upgrade to Webpack 4

### PR DESCRIPTION
**Pull request by klinker24**
I am working on #28. To get the web workers working through webpack, we needed to upgrade to webpack v4. This should do that.

@Serubin if you get a chance, can you try this out and make sure it is working for you in both dev/production? It seems to be working alright, but an extra set of eyes would be great.

### CODE REVIEW

File: build/webpack.prod.conf.js [hunk 1]
```diff
@@ -27,21 +28,9 @@ var webpackConfig = merge(baseWebpackConfig, {
...
-      },
-      output: {
-        ascii_only: true
-      },
-      sourceMap: true
-    }),
     // extract css into its own file
     new ExtractTextPlugin({
-      filename: utils.assetsPath('css/[name].[contenthash].css')
+      filename: utils.assetsPath('css/[name].[md5:contenthash:hex:20].css')
```
Comment (klinker24) [inline build/webpack.prod.conf.js]: https://github.com/webpack-contrib/extract-text-webpack-plugin/issues/763#issuecomment-377990665

File: build/webpack.base.conf.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ var utils = require('./utils')
 var config = require('../config')
 var vueLoaderConfig = require('./vue-loader.conf')
 var SWPrecacheWebpackPlugin = require('sw-precache-webpack-plugin')
+var VueLoaderPlugin = require('vue-loader/lib/plugin')
```
Comment (klinker24) [inline build/webpack.base.conf.js]: https://vue-loader.vuejs.org/migrating.html

File: build/webpack.prod.conf.js [hunk 3]
```diff
@@ -27,21 +28,9 @@ var webpackConfig = merge(baseWebpackConfig, {
   },
   plugins: [
     // http://vuejs.github.io/vue-loader/en/workflow/production.html
-    new webpack.DefinePlugin({
```
Comment (klinker24) [inline build/webpack.prod.conf.js]: https://webpack.js.org/migrate/4/

**Comment by Serubin** [comment]
This appears to be good!

**Comment by Serubin** [comment]
I did get a Deprecation Warning, but not a huge deal. 
```
(node:42993) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
```

**Comment by klinker24** [comment]
Yeah that warning is coming from one of the plugins that hadn't been updated for webpack 4 yet. Still working fine though, so I am not concerned about it at this point

**Pull request merged**